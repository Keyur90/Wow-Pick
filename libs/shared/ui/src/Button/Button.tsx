import React, { ReactNode } from 'react';
import { Button as MuiButton } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

interface ButtonPropTypes {
  children: NonNullable<ReactNode>;
  onClick: () => void;
  variant?: ButtonVariantType;
  fullWidth?: boolean;
  disabled?: boolean;
}

type ButtonVariantType = 'contained' | 'outlined';

const useStyles = makeStyles((theme) => ({
  outlinedPrimary: {
    background: theme.palette.common.white,
  },
}));

const Button: React.FC<ButtonPropTypes> = ({
  children,
  onClick,
  variant = 'contained',
  fullWidth = true,
  disabled = false,
}) => {
  const classes = useStyles();

  return (
    <MuiButton
      classes={classes}
      color="primary"
      fullWidth={fullWidth}
      variant={variant}
      size={'large'}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </MuiButton>
  );
};

export { Button };
