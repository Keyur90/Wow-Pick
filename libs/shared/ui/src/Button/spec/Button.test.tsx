import React from 'react';
import { render } from '@testing-library/react';
import { Button } from '../Button';

test('renders Primary Button', () => {
  const { container } = render(<Button onClick={() => null}>Start</Button>);

  expect(container).toMatchSnapshot();
});

test('renders Outlined Primary Button', () => {
  const { container, getByText } = render(<Button onClick={() => null}>Start</Button>);

  expect(container).toMatchSnapshot();
});
