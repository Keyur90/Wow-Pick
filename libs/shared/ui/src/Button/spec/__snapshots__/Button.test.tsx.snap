// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Outlined Primary Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Start
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`renders Primary Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Start
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
